modules/dbparser/tests/test_radix.c

```diff
@@ -32,199 +32,112 @@
 {
-  RNode *ret = r_find_node(root, key, strlen(key), NULL);
+  RNode *ret = r_find_node(root, (gchar *)key, strlen(key), NULL);
 
-  if (ret && expected_value)
-    {
-      if (strcmp(ret->value, expected_value) != 0)
-        {
-          printf("FAIL: returned value does not match expected: '%s' <> '%s'\n", (gchar *) ret->value, expected_value);
-          fail = TRUE;
-        }
-      else if (verbose)
-        {
-          printf("PASS: found expected: '%s' => '%s'\n", key, (gchar *) ret->value);
-        }
-    }
-  else if (ret && !expected_value)
+  if (expected_value)
     {
-      printf("FAIL: found unexpected: '%s' => '%s'\n", key, (gchar *) ret->value);
-      fail = TRUE;
+      cr_assert(ret, "node not found. key=%s\n", key);
+      cr_expect_str_eq(ret->value, expected_value, "FAIL: returned value does not match expected: '%s' <> '%s'\n",
+                       (gchar *) ret->value, expected_value);
     }
-  else if (expected_value)
+  else
     {
-      printf("FAIL: not found while expected: '%s' => none\n", key);
-      fail = TRUE;
+      cr_assert_not(ret, "found unexpected: '%s' => '%s'\n", key, (gchar *) ret->value);
     }
 }
 
 void
-test_search(RNode *root, gchar *key, gboolean expect)
+test_search(RNode *root, const gchar *key, gboolean expect)
 {
   test_search_value(root, key, expect ? key : NULL);
 }
 
 void
-test_search_matches(RNode *root, gchar *key, gchar *name1, ...)
+test_search_matches(RNode *root, const gchar *key, const gchar *search_pattern[])
 {
-  RNode *ret;
-  va_list args;
   GArray *matches = g_array_new(FALSE, TRUE, sizeof(RParserMatch));
+  g_array_set_size(matches, 1);
+
   RParserMatch *match;
   const gchar *match_name;
 
-  g_array_set_size(matches, 1);
-  va_start(args, name1);
+  RNode *ret = r_find_node(root, (guint8 *)key, strlen(key), matches);
 
-  ret = r_find_node(root, key, strlen(key), matches);
-  if (ret && !name1)
+  if (!search_pattern[0])
     {
-      printf("FAIL: found unexpected: '%s' => '%s' matches: ", key, (gchar *) ret->value);
-      for (gsize i = 0; i < matches->len; i++)
-        {
-          match = &g_array_index(matches, RParserMatch, i);
-          match_name = log_msg_get_value_name(match->handle, NULL);
-          if (match_name)
-            {
-              if (!match->match)
-                printf("'%s' => '%.*s'", match_name, match->len, &key[match->ofs]);
-              else
-                printf("'%s' => '%s'", match_name, match->match);
-            }
-        }
-      printf("\n");
-      fail = TRUE;
+      cr_expect_not(ret, "found unexpected: '%s' => '%s' matches: ", key, (gchar *) ret->value);
     }
-  else if (ret && name1)
+  else
     {
-      gsize i = 1;
-      gchar *name, *value;
+      cr_assert(ret, "not found while expected: '%s' => none %s\n", key, search_pattern[0]);
 
-      name = name1;
-      value = va_arg(args, gchar *);
-      while (name)
+      for (int i=0; search_pattern[i]; i+=2)
```

Comment:
For future refactor idea: the name values could be put into a structure, thus avoid the +=2 complexity.